File: ios/CareDoc/Sources/CareDoc/Features/CalendarScreen.swift
import SwiftUI

/// 畫面 8：行事曆
struct CalendarScreen: View {
    @Bindable var state: AppState
    @SwiftUI.State private var selectedDay = 13

    private let today = 13
    private let weekdays = ["日", "一", "二", "三", "四", "五", "六"]
    // 2026 年 6 月：6/1 是週一
    private let leadingBlanks = 1
    private let daysInMonth = 30

    private var eventDays: Set<Int> {
        Set(state.session.events.compactMap { $0.kind == .medication ? nil : $0.date.day })
    }
    private var medDays: Set<Int> {
        Set(state.session.events.compactMap { $0.kind == .medication ? $0.date.day : nil })
    }
    private var dayEvents: [ScheduleEvent] {
        state.session.events.filter { $0.date.day == selectedDay || $0.kind == .medication }
    }

    var body: some View {
        ScrollView(showsIndicators: false) {
            VStack(alignment: .leading, spacing: 14) {
                PageHeader(kicker: "2026 年", title: "6 月")

                // 月曆
                VStack(spacing: 6) {
                    HStack {
                        ForEach(weekdays, id: \.self) { d in
                            Text(d).font(.cdBody(10, weight: .heavy))
                                .foregroundStyle(CD.text3)
                                .frame(maxWidth: .infinity)
                        }
                    }
                    let columns = Array(repeating: GridItem(.flexible(), spacing: 4), count: 7)
                    LazyVGrid(columns: columns, spacing: 4) {
                        ForEach(0..<leadingBlanks, id: \.self) { _ in Color.clear.frame(height: 34) }
                        ForEach(1...daysInMonth, id: \.self) { day in
                            dayCell(day)
                        }
                    }
                }
                .card(padding: 12)

                SectionHeader(title: "6/\(selectedDay) \(weekdayName(selectedDay))",
                              trailing: "今天") {
                    Haptics.shared.selectionTick()
                    withAnimation(CD.ease) { selectedDay = today }
                }

                if dayEvents.isEmpty {
                    Text("這天沒有安排，休息也是照護的一部分")
                        .font(.cdBody(13, weight: .medium)).foregroundStyle(CD.text2)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 26)
                        .card()
                } else {
                    ForEach(dayEvents) { event in
                        EventRow(event: event)
                    }
                }

                PillButton(title: "加入手機行事曆 (.ics)", icon: "square.and.arrow.down", style: .accent) {
                    // POC：永遠免費的功能，真實版產出 .ics
                }
                Text(".ics 匯出永遠免費")
                    .font(.cdBody(11, weight: .medium)).foregroundStyle(CD.text3)
                    .frame(maxWidth: .infinity)

                Spacer(minLength: 100)
            }
            .padding(.horizontal, 18)
            .padding(.top, 12)
        }
    }

    private func dayCell(_ day: Int) -> some View {
        Button {
            Haptics.shared.selectionTick()
            withAnimation(CD.ease) { selectedDay = day }
        } label: {
            VStack(spacing: 2) {
                Text("\(day)")
                    .font(.cdBody(12.5, weight: day == selectedDay ? .heavy : .bold))
                Circle()
                    .fill(dotColor(day))
                    .frame(width: 4, height: 4)
            }
            .frame(maxWidth: .infinity)
            .frame(height: 34)
            .background(
                day == selectedDay ? CD.accent : .clear,
                in: RoundedRectangle(cornerRadius: 9, style: .continuous)
            )
            .foregroundStyle(day == selectedDay ? CD.plumDeep : CD.text)
        }
        .buttonStyle(.plain)
    }

    private func dotColor(_ day: Int) -> Color {
        if eventDays.contains(day) { return CD.info }
        if medDays.contains(day) { return CD.accent }
        return .clear
    }

    private func weekdayName(_ day: Int) -> String {
        // 2026/6/1 週一 → index = day % 7
        weekdays[(day % 7 + 0) % 7].isEmpty ? "" : "週" + weekdays[(day + 0) % 7]
    }
}

File: flutter/carrius/lib/screens/calendar_screen.dart
import 'package:flutter/material.dart';
import '../design/tokens.dart';
import '../design/components.dart';
import '../design/haptics.dart';
import '../models/models.dart';
import '../state/app_state.dart';
import 'home_screen.dart' show EventRow;

class CalendarScreen extends StatefulWidget {
  final AppState state;
  const CalendarScreen({super.key, required this.state});

  @override
  State<CalendarScreen> createState() => _CalendarScreenState();
}

class _CalendarScreenState extends State<CalendarScreen> {
  static const _weekdays = ['日', '一', '二', '三', '四', '五', '六'];
  static const _today = 13;
  static const _leadingBlanks = 1; // 2026/6/1 = 週一
  static const _daysInMonth = 30;
  int _selected = 13;

  Set<int> get _eventDays =>
      widget.state.session.events.where((e) => e.kind != EventKind.medication).map((e) => e.day).toSet();
  Set<int> get _medDays =>
      widget.state.session.events.where((e) => e.kind == EventKind.medication).map((e) => e.day).toSet();
  List<ScheduleEvent> get _dayEvents =>
      widget.state.session.events.where((e) => e.day == _selected || e.kind == EventKind.medication).toList();

  @override
  Widget build(BuildContext context) {
    final p = PaletteScope.of(context);
    return ListView(
      padding: const EdgeInsets.fromLTRB(18, 12, 18, 120),
      children: [
        PageHeader(kicker: '2026 年', title: '6 月'),
        const SizedBox(height: 14),
        CDCard(
          padding: const EdgeInsets.all(12),
          child: Column(
            children: [
              Row(
                children: [
                  for (final d in _weekdays)
                    Expanded(child: Center(child: Text(d, style: CDText.body(10, weight: FontWeight.w900, color: p.text3)))),
                ],
              ),
              const SizedBox(height: 6),
              GridView.count(
                crossAxisCount: 7,
                shrinkWrap: true,
                physics: const NeverScrollableScrollPhysics(),
                mainAxisSpacing: 4,
                crossAxisSpacing: 4,
                childAspectRatio: 1,
                children: [
                  for (var i = 0; i < _leadingBlanks; i++) const SizedBox.shrink(),
                  for (var day = 1; day <= _daysInMonth; day++) _dayCell(p, day),
                ],
              ),
            ],
          ),
        ),
        const SizedBox(height: 14),
        SectionHeader(title: '6/$_selected ${_weekdayName(_selected)}', trailing: '今天', onTap: () {
          Haptics.selectionTick();
          setState(() => _selected = _today);
        }),
        const SizedBox(height: 10),
        if (_dayEvents.isEmpty)
          CDCard(
            child: Center(
              child: Padding(
                padding: const EdgeInsets.symmetric(vertical: 18),
                child: Text('這天沒有安排，休息也是照護的一部分', style: CDText.body(13, weight: FontWeight.w500, color: p.text2)),
              ),
            ),
          )
        else
          for (final e in _dayEvents) ...[EventRow(event: e), const SizedBox(height: 10)],
        const SizedBox(height: 4),
        PillButton(
            title: '加入手機行事曆 (.ics)',
            icon: Icons.ios_share,
            onTap: () => showComingSoon(context, '.ics 匯出')),
        const SizedBox(height: 8),
        Center(child: Text('.ics 匯出永遠免費，正式版開放', style: CDText.body(11, weight: FontWeight.w500, color: p.text3))),
      ],
    );
  }

  Widget _dayCell(Palette p, int day) {
    final selected = day == _selected;
    Color dot = Colors.transparent;
    if (_eventDays.contains(day)) dot = CD.info;
    if (_medDays.contains(day)) dot = CD.accent;
    return GestureDetector(
      onTap: () {
        Haptics.selectionTick();
        setState(() => _selected = day);
      },
      child: Container(
        decoration: BoxDecoration(
          color: selected ? CD.accent : Colors.transparent,
          borderRadius: BorderRadius.circular(9),
        ),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Text('$day',
                style: CDText.body(12.5, weight: selected ? FontWeight.w900 : FontWeight.w700, color: selected ? CD.plumDeep : p.text)),
            const SizedBox(height: 2),
            Container(width: 4, height: 4, decoration: BoxDecoration(color: dot, shape: BoxShape.circle)),
          ],
        ),
      ),
    );
  }

  String _weekdayName(int day) => '週${_weekdays[day % 7]}';
}
